docs/usage.rst

```diff
@@ -0,0 +1,259 @@
+The same applies for using the management command::
+
+    python manage.py pgactivity ls -f context__key=value
+
+Next are ways you can automatically attach context from
+requests, management commands,
+and background tasks.
+
+Requests
+~~~~~~~~
+
+We recomend adding the `pgactivity.middleware.ActivityMiddleware`
+middleware to ``settings.MIDDLEWARE``. This will automatically track
+both the ``url`` and ``method`` for every request, allowing you to see which
+URL issued a query when viewing it. This can be helpful when determining
+if it's safe to kill a particular query.
+
+Management Commands
+~~~~~~~~~~~~~~~~~~~
+
+One-off management commands that don't go through requests can
+also be instrumented in the ``manage.py`` file of a project
+using `pgactivity.contrib.execute_from_command_line`:
+
+.. code-block:: python
+
+    #!/usr/bin/env python
+    import os
+    import sys
+
+
+    if __name__ == "__main__":
+        os.environ.setdefault("DJANGO_SETTINGS_MODULE", "settings")
+        from pgactivity.contrib import execute_from_command_line
+
+        execute_from_command_line(sys.argv)
+
+This function is a wrapper of Django's ``execute_from_command_line``
+and the ``command`` key added as context.
+It ignores ``runserver`` and ``runserver_plus`` by default.
+
+If it's not possible to use `pgactivity.contrib.execute_from_command_line`
+in your ``manage.py``, you can achieve the same functionality like so:
+
+.. code-block:: python
+
+    import contextlib
+    import sys
+    from django.core.management import execute_from_command_line
+
+
+    ignore_commands = ["runserver", "runserver_plus"]
+
+    if len(sys.argv) > 1 and not sys.argv[1] in ignore_commands:
+        activity_context = runtime.context(command=sys.argv[1])
+    else:
+        activity_context = contextlib.ExitStack()
+
+    with activity_context:
+        execute_from_command_line(sys.argv)
+
+Celery Tasks
+~~~~~~~~~~~~
+
+Celery tasks can also be instrumented like so:
+
+.. code-block:: python
+
+    import celery
+    import pgactivity
+
+    class Task(celery.Task):
+        def __call__(self, *args, **kwargs):
+            with pgactivity.context(task=self.name):
+                return super().__call__(*args, **kwargs)
+
+
+    # Override the celery task decorator for your application
+    app = create_celery_app('my-app')
+    task = app.task(base=Task)
+
+Setting Statement Timeouts
+--------------------------
+
+Using the `pgactivity.timeout` decorator and context manager to dynamically
+set Postgres's ``statement_timeout`` setting.
+
+For example, the following code ensures that no single query takes longer than
+500 milliseconds:
+
+.. code-block:: python
+
+    import pgactivity
+
+    with pgactivity.timeout(pgactivity.timedelta(milliseconds=500)):
+        # Issue queries. Any that exceed 500 milliseconds will raise an exception
+
+.. note::
+
+    `pgactivity.timedelta` is just a shortcut for Python's `datetime.timedelta`.
```

Comment:
I will change this so that it just accepts the same keyword args as `timedelta`. Then users won't have to worry about doing conversions to different time units